### PR TITLE
feat(external-webui-redis): Add the possibility to use an external Redis Database

### PR DESCRIPTION
**Pull request by Lowaiz**
Signed-off-by: Lyonel Martinez <lyonel.martinez@numberly.com>

**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

> /kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

/kind feature

> If this PR will release a new chart version please make sure to also uncomment the following line:

/kind chart-release

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area falco-chart

> /area falco-exporter-chart

 /area falcosidekick-chart

**What this PR does / why we need it**:

Add the options to use an external database for the webui.
Now the Redis' values have an `enabled` member and the Webui's values have an `externalRedis` member, containing an `enabled`, an `url` and a `port` member.

**Which issue(s) this PR fixes**:

None

**Special notes for your reviewer**:

**Checklist**

- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] CHANGELOG.md updated

### CODE REVIEW

**Comment by Issif** [comment]
Thanks. I will review in details asap. Can you just add in the descriptions of settings that the redis must have the redissearch module enabled, it's a requirement. Thanks

File: falcosidekick/CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,11 @@ numbering uses [semantic versioning](http://semver.org).
 Before release 0.1.20, the helm chart can be found in `falcosidekick` [repository](https://github.com/falcosecurity/falcosidekick/tree/master/deploy/helm/falcosidekick).
 
 
+## 0.5.5
+
+* Add `webui.externalRedis` with `enabled`, `url` and `port` to values to set an external Redis database with RediSearch > v2 for the WebUI
+* Add `webui.redis.enabled` option to disable the deployment of the database.
+
 ## 0.5.3
```
Comment (Issif) [inline falcosidekick/CHANGELOG.md:14]:
```suggestion
## 0.5.4
```
I forgot to change this version number after my last copy paste. Can you update it for me please.

File: falcosidekick/templates/deployment-ui.yaml [hunk 2]
```diff
@@ -49,7 +49,13 @@ spec:
           imagePullPolicy: {{ .Values.webui.image.pullPolicy }}
           args:
             - "-r"
+            {{- if .Values.webui.redis.enabled }}
             - {{ include "falcosidekick.fullname" . }}-ui-redis{{ if .Values.webui.redis.fullfqdn }}.{{ .Release.Namespace }}.svc.cluster.local{{ end }}:{{ .Values.webui.redis.service.port | default "6379" }}
+            {{- else if .Values.webui.externalRedis.enabled }}
+            - "{{ required "External Redis is enable. Please set the URL to the database." .Values.webui.externalRedis.url }}:{{ .Values.webui.externalRedis.port | default "6379" }}"
```
Comment (Issif) [inline falcosidekick/templates/deployment-ui.yaml:55]:
```suggestion
            - "{{ required "External Redis is enabled. Please set the URL to the database." .Values.webui.externalRedis.url }}:{{ .Values.webui.externalRedis.port | default "6379" }}"
```

**Review by Issif — CHANGES_REQUESTED** [review]
I'm thinking we could simplify a little the UX by removing the `webui.externalRedis.enabled` and `webui.redis.enabled`.
The conditions could be only:
- if `webui.externalRedis.url != ""` then disable the local redis

wdyt?

**Comment by Lowaiz** [comment]
> I just think we need a way to make them exclusive
An exclusive test now is done in the beginning of the deployment-ui file.

File: falcosidekick/templates/deployment-ui.yaml [hunk 3]
```diff
@@ -1,4 +1,9 @@
 {{- if .Values.webui.enabled }}
+{{- if and .Values.webui.redis.enabled .Values.webui.externalRedis.enabled }}
+  {{ fail "Both webui.redis and webui.externalRedis modules are enabled. Please unset one them." }}
```
Comment (Issif) [inline falcosidekick/templates/deployment-ui.yaml:3]:
```suggestion
  {{ fail "Both webui.redis and webui.externalRedis modules are enabled. Please disable one of them." }}
```

**Review by Issif — CHANGES_REQUESTED** [review]

**Review by Issif — APPROVED** [review]

**Pull request merged**